Automation/PowerShell/Automated AD User Password Change.ps1

```diff
@@ -0,0 +1,94 @@
+    $characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789/*-+.'
+    $randomString = ""
+    for ($i = 0; $i -lt 20; $i++) {
+        $randomString += $characters[(Get-Random -Minimum 0 -Maximum $characters.Length)]
+    }
+    return $randomString
+}
+
+
+# Set the Configuration for the Documents store
+$Password = ConvertTo-SecureString "PASSWORD" -AsPlainText -Force
+$Username = "automated_task"
+$RoyalStore = New-RoyalStore -UserName $Username
+
+# Specify the base directory where GUID-named subdirectories are located
+$baseDirectory = "C:\RoyalServer\DocumentStore\Documents"
+
+# Get a list of subdirectories with GUID names
+$subdirectories = Get-ChildItem -Path $baseDirectory -Directory
+
+
+    # Construct the path to the default.rtsz file in the current subdirectory
+    $documentPath = "C:\RoyalServer\DocumentStore\Documents\ID\default.rtsz"
+
+
+        # Open the Royal Document
+        $RoyalDocument = Open-RoyalDocument -FileName $documentPath -Store $RoyalStore -Password $Password
+
+        # Get the credentials from the Royal TS document
+        $credentials = Get-RoyalObject -Store $RoyalStore -Type RoyalCredential
+
+        # Find the Domain Administrator credential object and store the login details
+        $domainAdminCredential = $credentials | Where-Object { $_.Name -eq "Domain Administrator" }
+        $domainAdminCredential = New-Object System.Management.Automation.PSCredential ($domainAdminCredential.UserName, ($domainAdminCredential.Password | ConvertTo-SecureString -AsPlainText -Force))
+
+        # Loop through each credential object
+        foreach ($credential in $credentials) {
+            if ($credential.CustomField3 -eq "msad") {
+                Write-Host "found"
+                # Generate a random 20-character string
+                $newPassword = Generate-RandomString
+                $secureNewPassword = ConvertTo-SecureString -String $newPassword -AsPlainText -Force
+
+                # Set the AD user password
+                $sessionOption = New-PSSessionOption -SkipCACheck -SkipCNCheck
+                try {
+                    Invoke-Command -ComputerName $ComputerName -Credential $domainAdminCredential -UseSSL -SessionOption $sessionOption -ScriptBlock {
+                        param($adUserName, $secureNewPassword)
+                        Set-ADAccountPassword -Identity $adUserName -NewPassword $secureNewPassword -Reset
+                    } -ArgumentList $credential.Name, $secureNewPassword
+
+                    # Store the new password in the credential object
+                    $credential.Password = $newPassword
+                    Write-Host "Password updated successfully for AD user $($credential.Name)."
+                } catch {
+                    Write-Host "Failed to update password for AD user $($credential.Name): $_"
+                    Write-Host "Failed to update password for AD user $($credential.Name): $_"
+            }
+        }
```

Comment:
- Second identical `Write-Host` seems redundant
- Could you please fix the indentation of the curly braces